tests/data/test_checkpoint.py

```diff
@@ -0,0 +1,129 @@
+
+from elfpy.data import postgres
+from elfpy.data.db_schema import Base, CheckpointInfo
+
+engine = create_engine("sqlite:///:memory:")  # in-memory SQLite database for testing
+Session = sessionmaker(bind=engine)
+
+# fixture arguments in test function have to be the same as the fixture name
+# pylint: disable=redefined-outer-name
+
+
+@pytest.fixture(scope="function")
+def session():
+    """Session fixture for tests"""
+    Base.metadata.create_all(engine)  # create tables
+    session_ = Session()
+    yield session_
+    session_.close()
+    Base.metadata.drop_all(engine)  # drop tables
+
+
+class TestCheckpointTable:
+    """CRUD tests for checkpoint table"""
+
+    def test_create_checkpoint(self, session):
+        """Create and entry"""
+        # Note: this test is using inmemory sqlite, which doesn't seem to support
+        # autoincrementing ids without init, whereas postgres does this with no issues
+        # Hence, we explicitly add id here
+        timestamp = datetime.now()
+        checkpoint = CheckpointInfo(blockNumber=1, timestamp=timestamp)
+        postgres.add_checkpoint_infos([checkpoint], session)
+        session.commit()
+
+        retrieved_checkpoint = session.query(CheckpointInfo).filter_by(blockNumber=1).first()
+        assert retrieved_checkpoint is not None
+        # event_value retreieved from postgres is in Decimal, cast to float
+        assert retrieved_checkpoint.timestamp == timestamp
+
+    def test_update_checkpoint(self, session):
+        """Update an entry"""
+        timestamp = datetime.now()
+        checkpoint = CheckpointInfo(blockNumber=1, timestamp=timestamp)
+        postgres.add_checkpoint_infos([checkpoint], session)
+        session.commit()
+
+        checkpoint.sharePrice = 5.0
+        session.commit()
+
+        updated_checkpoint = session.query(CheckpointInfo).filter_by(blockNumber=1).first()
+        # event_value retreieved from postgres is in Decimal, cast to float
+        assert updated_checkpoint.sharePrice == 5.0
+
+    def test_delete_checkpoint(self, session):
+        """Delete an entry"""
+        timestamp = datetime.now()
+        checkpoint = CheckpointInfo(blockNumber=1, timestamp=timestamp)
+        postgres.add_checkpoint_infos([checkpoint], session)
+        session.commit()
+
+        session.delete(checkpoint)
+        session.commit()
+
+        deleted_checkpoint = session.query(CheckpointInfo).filter_by(blockNumber=1).first()
+        assert deleted_checkpoint is None
+
+
+class TestCheckpointInterface:
+    """Testing postgres interface for checkpoint table"""
+
+    def test_latest_block_number(self, session):
+        """Testing retrevial of checkpoint via interface"""
+        checkpoint_1 = CheckpointInfo(blockNumber=1, timestamp=datetime.now())
+        postgres.add_checkpoint_infos([checkpoint_1], session)
+        session.commit()
+
+        latest_block_number = postgres.get_latest_block_number_from_table(CheckpointInfo.__tablename__, session)
```

Comment:
Don't forget to fix this call to `get_latest_block_number_from_table` after you rebase from main